tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-table/attributelist-table.component.ts

```diff
@@ -215,6 +245,110 @@ export class AttributelistTableComponent implements AttributelistTable, OnInit,
+      if (data.success) {
+        let uniqueValues: FilterValueSettings[];
+        uniqueValues = [];
+        const colObject = this.layerFilterValues.columns.find(c => c.name === columnName);
+        const colIndex = this.layerFilterValues.columns.findIndex(obj => obj.name === columnName);
+        if (colObject.uniqueValues.length === 0) {
+
+          data.uniqueValues[columnName].forEach(val => {
+            let filterValueSettings: FilterValueSettings;
+            filterValueSettings = {value: val, select: true};
+            uniqueValues.push(filterValueSettings);
+          })
+        } else {
+            colObject.uniqueValues.forEach(val => uniqueValues.push(Object.assign({}, val)))
+        }
+
+        const config = new MatDialogConfig();
+        config.data = {
+          colName: columnName,
+          values: uniqueValues,
+        };
+        const dialogRef = this.dialog.open(AttributelistFilterValuesFormComponent, config);
+        dialogRef.afterClosed().subscribe(filterSetting => {
+          // Do the filtering
+          if (filterSetting !== 'CANCEL') {
+            if (filterSetting === 'ON') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = config.data.values;
+              this.layerFilterValues.columns[colIndex].nullValue = false;
+              this.layerFilterValues.columns[colIndex].status = true;
+            } else if (filterSetting === 'NONE') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = config.data.values;
+              this.layerFilterValues.columns[colIndex].nullValue = true;
+              this.layerFilterValues.columns[colIndex].status = true;
+            } else if (filterSetting === 'OFF') {
+              this.layerFilterValues.columns[colIndex].uniqueValues = [];
+              this.layerFilterValues.columns[colIndex].nullValue = false;
+              this.layerFilterValues.columns[colIndex].status = false;
+            }
+            this.createFilter();
+            this.updateTable();
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Create the CQL filter string
+   */
+  public createFilter(): void {
+    this.valueFilter = '';
+    let filteredColumns = 0;
+    this.layerFilterValues.columns.forEach((c) => {
+      if (c.status) {
+        filteredColumns++;
+        if (filteredColumns === 1) {
+          this.valueFilter = ' ';
+        } else {
+          this.valueFilter += ' AND';
+        }
+        if (c.nullValue) {
+          this.valueFilter += ' ' + c.name + ' IS NULL';
+        } else {
+          this.valueFilter += ' ' + c.name + ' IN (';
+          let filteredValues = 0;
+          c.uniqueValues.forEach((v) => {
+            if (v.select) {
+              filteredValues++;
+              if (filteredValues > 1) {
+                this.valueFilter += ',';
+              }
+              this.valueFilter += '\'' + v.value + '\'';
```

Comment:
Volgens mij gaat dit niet goed met numerieke waardes: die moeten niet gequote worden

Reply:
Dat dacht ik ook, maar tot mijn verbazing ging het wel goed. Ook het filteren op bij. de numeriek velden jaartal en oppervlakte ging gewoon goed.

Reply:
In de toekomst zou dit wel fout kunnen gaan als er strikter omgegaan wordt met CQL.
Voor de zekerheid alleen quotes bij numerieke waarden gezet.